src/greenlight/runner.clj

```diff
@@ -77,6 +81,60 @@
     true))
 
 
+(defmacro ^:private with-delayed-output
+  [printer & body]
+  `(let [out# (java.io.StringWriter.)
+         original-out# *out*
+         original-err# *err*]
+     (binding [*out* out#, *err* out#]
+       (let [result# ~@body]
+         (binding [*out* original-out#
+                   *err* original-err#]
+           (~printer (str out#))
```

Comment:
Macro best practice: let `printer# ~printer` earlier in the macro, so that if someone passes in a form that _constructs_ a printer function, you aren't evaluating it inside these bindings.

Reply:
I removed this delayed output printer for a piped reader/writer that allows streaming one test's output at a time, instead of waiting for test completion to print anything.